Blockzilla/OverlayView.swift

```diff
@@ -304,7 +303,9 @@ class OverlayView: UIView {
     }
 
     @objc private func didPressSearch(sender: IndexedInsetButton) {
-        delegate?.overlayView(self, didSearchForQuery: searchSuggestions[sender.getIndex()])
+        // Safeguard against a potential update of searchSuggestions causing this index to no longer exist
```

Comment:
If I were to use
```    
         guard sender.getIndex() < searchSuggestions.count
         delegate?.overlayView(self, didSearchForQuery: searchSuggestions[sender.getIndex()])
```
it's possible that searchSuggestions gets updated *after* the guard but before the indexing on the second line. Because of this, I believe that making an immutable copy is the safest approach. Please let me know if you disagree!